### ISSUE TITLE
Unable to find IPv4 address for ORPort 9001

### ISSUE THREAD

**Issue by andrewbpowers** [DarkIsDude/tor-server#6]
Hello!

I got this Problem:
`Nov 17 08:34:44.000 [notice] Unable to find IPv4 address for ORPort 9001. You might want to specify IPv4Only to it or set an explicit address or set Address.`

This entry doesn't work in the ./torrc file:
`ORPort 9001 IPv4Only`

Idea for a solution to the problem?

Thank you and take care...

**Comment by DarkIsDude** [comment]
What do you mean by `ORPort 9001 IPv4Only` doesn't work ? I'm running it and working as expected. Maybe more a network/tor issue ?

**Comment by andrewbpowers** [comment]
To make a long story short... "chriswayg/tor-server" version worked immediately and without any problems.
But not your version!

`Unable to find IPv4 address for ORPort 9001. You might want to specify IPv4Only to it or set an explicit address or set Address.`

The error is displayed with `ORPort 9001`, what your default configuration is in the torrc file and also with the option `IPv4Only`.

**Comment by DarkIsDude** [comment]
Which command are you using ? Do you have a special configuration related to the docker container ?

`chriswayg/tor-server` version is really old (I think 0.2.x.x) and we are now 0.4.x.x. . A lot of changes had beed done regarding TOR configuration file and you'll maybe need to update some stuff :/

**Comment by andrewbpowers** [comment]
Ok… again…
I run your default config:
`docker run -d --name=tor_relay_1 -p 9001:9001 --restart=always doudou34/tor-server `
It also doesn’t work when I install it with Portainer.

`chriswayg/tor-server ` runs perfectly, but unfortunately it’s a old Tor version.

**Comment by andrewbpowers** [comment]
With the latest version of `chriswayg/tor-server` the Tor server runs without any problem and immediately and with a few adjustments also a Nginx behind it worked. Almost out of the box!

With your version I had to manually assign port 9030, 80, 443, etc in docker too. 

If I adjust the torrc file from your version so that it is identical to the working `chriswayg/tor-server` config, then the Tor server does not work together with the Ngnix server either.

As I understand it, your version should simply be an always up-to-date version of `chriswayg/tor-server`.
But this is apparently not the case.

> `chriswayg/tor-server` version is really old (I think 0.2.x.x) and we are now 0.4.x.x. . A lot of changes had beed done regarding TOR configuration file and you'll maybe need to update some stuff :/

Content of the torrc config file has changed very little.

**Comment by DarkIsDude** [comment]
As I said, there are a lot of changes. My goal is indeed to keep `chriswayg/tor-server` up to date. But the way that `chriswayg` is using to install Tor is not compatible with the latest version. So needed to change some stuff. You can check commit if you want but it's more or less the same process.

I tried the same command as you and it's working perfectly well so maybe it's more a configuration issue ? Did you find how to fix it ?

**Comment by andrewbpowers** [comment]
Thanks for the feedback! :-)

In fact, some things have changed under the hood.

In the meantime I have tested half a dozen images and most of them do not work because they are outdated or the configuration is simply wrong or wrongly taken over from others.

It looks like the option `IPv4Only`for the `ORPort 9001 `configuration has no effect on your version.
But I have read that the port 9001 is not that important anymore.

In the meantime I have created a few images myself and so far it works.
It was important to me that the "tor server" runs as host and communicates with the Ngnix.
Also there should be a torrc config file and a shell, for easy, clear and targeted customization.
